e2e/nomostest/clean.go

```diff
@@ -505,7 +505,19 @@ func DeleteObjectsAndWait(nt *NT, objs ...client.Object) error {
 		nn := client.ObjectKeyFromObject(obj)
 		gvk, err := kinds.Lookup(obj, nt.Scheme)
 		if err != nil {
-			return err
+			tg.Go(func() error {
+				return err
+			})
+			continue
+		}
+		// Remove fake test finalizers if they are blocking deletion
+		if reflect.DeepEqual(obj.GetFinalizers(), []string{ConfigSyncE2EFinalizer}) {
```

Comment:
I see why you added it here, but this really feels like a test specific cleanup. Like, the test should be in charge of registering it. 

Is there any way we could make the tests all register cleanup, even if they aren't running, then run all the cleanup first before any tests...?

Reply:
There may be a way to hack something like that together, but I think it would be a pretty large refactor to make the test scaffolding work in that way.

Currently [the test environments are set up and `Clean` is run before any of the test cases are invoked](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testcases/main_test.go#L131), so there is no knowledge of the test cases themselves.